site/app/libraries/FileUtils.php

```diff
@@ -519,14 +568,19 @@ public static function validateUploadedFiles($files) {
                 $err_msg = "Invalid filename";
             }
 
-            $success = $err_msg === "No error.";
+            //if zip file check files inside
+            if($is_zip && !FileUtils::checkFileInZipName($tmp_name) ){
+                $err_msg = "Invalid filename within zip file"; 
+            }
 
+            $success = $err_msg === "No error.";
             $ret[] = [
                 'name' => $name,
                 'type' => $type,
                 'error' => $err_msg,
                 'size' => $size,
-                'success' => $success
+                'success' => $success,
+                'is_zip' => $is_zip
             ];
         }
```

Comment:
I should have caught this in my initial review, sorry for not.

This function should be refactored so that we either make it clearer what the final value of `$err_msg` is going to be or collect all possible errors in an array and return them all. Personally, I'd vote for the latter just so the uploader can fix them all in one fell swoop instead of upload, see a new error, upload, see a new error, etc. until they finally after several rounds fix all errors.